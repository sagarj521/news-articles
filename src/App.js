import React, { Fragment } from 'react';
import './App.css';
import ArticleList from './containers/ArticleList';
import ArticleDetail from './components/ArticleDetail';
import { Route, Routes } from 'react-router';

function App() {
	return (
		<Fragment>
			<div className="App">
				<h1>NY Times Most Popular Articles</h1>
			</div>
			<Routes>
				<Route path="/" element={<ArticleList />} />
				<Route path="/news-detail" element={<ArticleDetail />} />
			</Routes>
			;
		</Fragment>
	);
}

export default App;
