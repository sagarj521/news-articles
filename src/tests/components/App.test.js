import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from '../../App';

// Mock the ArticleList and ArticleDetail components
jest.mock('../../containers/ArticleList', () => {
	const ArticleList = () => <div>Article List Component</div>;

	return ArticleList;
});

jest.mock('../../components/ArticleDetail', () => {
	const ArticleDetail = () => <div>Article Detail Component</div>;

	return ArticleDetail;
});

describe('App', () => {
	test('renders title correctly', () => {
		render(
			<MemoryRouter>
				<App />
			</MemoryRouter>
		);
		expect(screen.getByText('NY Times Most Popular Articles')).toBeInTheDocument();
	});

	test('renders ArticleList component for the root route', () => {
		render(
			<MemoryRouter initialEntries={['/']}>
				<App />
			</MemoryRouter>
		);
		expect(screen.getByText('Article List Component')).toBeInTheDocument();
	});

	test('renders ArticleDetail component for the news-detail route', () => {
		render(
			<MemoryRouter initialEntries={['/news-detail']}>
				<App />
			</MemoryRouter>
		);
		expect(screen.getByText('Article Detail Component')).toBeInTheDocument();
	});
});
