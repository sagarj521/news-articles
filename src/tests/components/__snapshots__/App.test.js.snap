// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      NY Times Most Popular Articles
    </h1>
  </div>
  <ul
    class="article-list"
  >
    <div
      class="loading"
      role="presentation"
    />
  </ul>
  ;
</DocumentFragment>
`;

exports[`ArticleDetail matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      NY Times Most Popular Articles
    </h1>
  </div>
  <ul
    class="article-list"
  >
    <div
      class="loading"
      role="presentation"
    />
  </ul>
  ;
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      NY Times Most Popular Articles
    </h1>
  </div>
  <ul
    class="article-list"
  >
    <div
      class="loading"
      role="presentation"
    />
  </ul>
  ;
</DocumentFragment>
`;
